kore/app/parser/Main.hs

```diff
@@ -80,43 +80,33 @@ main = handleTop $ do
             Nothing  -- environment variable name for extra arguments
             parseKoreParserOptions
             parserInfoModifiers
-    for_ (localOptions options) $ \koreParserOptions ->
-        flip runLoggerT Log.emptyLogger $ do
+    for_ (localOptions options) $ \koreParserOptions -> runEmptyLogger $ do
+        indexedModules <- do
             let KoreParserOptions { fileName } = koreParserOptions
             parsedDefinition <- mainDefinitionParse fileName
             let KoreParserOptions { willVerify } = koreParserOptions
-            indexedModules <- if willVerify
-                then lift $ mainVerify parsedDefinition
-                else return Map.empty
+            indexedModules <-
+                if willVerify
+                then fmap Just . lift $ mainVerify parsedDefinition
+                else pure Nothing
             let KoreParserOptions { willPrintDefinition } = koreParserOptions
             let KoreParserOptions { appKore } = koreParserOptions
-            lift $ when willPrintDefinition
-                $ if appKore
-                    then putStrLn
-                        $ unparseToString2
-                        $ completeDefinition
-                        $ toVerifiedDefinition indexedModules
-                else putDoc (debug parsedDefinition)
-
-            let KoreParserOptions { patternOpt } = koreParserOptions
-            for_ patternOpt $ \patternOptions -> do
-                let PatternOptions { mainModuleName } = patternOptions
-                indexedModule <- if willVerify then Just <$>
-                    ( lookupMainModule
-                        (ModuleName mainModuleName)
-                        indexedModules
-                    & lift )
-                    else return Nothing
-                let PatternOptions { patternFileNames } = patternOptions
-                for_ patternFileNames $ \patternFileName -> do
-                    parsedPattern <- mainPatternParse patternFileName
-                    when willVerify $ do
-                        _ <- mainPatternVerify (fromJust indexedModule) parsedPattern
-                        return ()
-                    let KoreParserOptions { willPrintPattern } =
-                            koreParserOptions
-                    when willPrintPattern $
-                        lift $ putDoc (debug parsedPattern)
+            when willPrintDefinition $ putDebug parsedDefinition
+            when appKore $ for_ indexedModules printAppKore
```

Comment:
@ttuegel 
I see that the logic changed here, in the sense that we may print both the applicative Kore of the modules AND the definition.
Is this what we want?
Isn't the idea of the `appKore` flag to change what is printed if `willPrintDefinition` is also set?

Reply:
Yes, the new behavior is nonsensical, but the old behavior was nonsensical also. If you want to change `when willPrintDefinition` to `when (willPrintDefinition && not appKore)`, that's fine with me.